bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/rest/internal/HABotResource.java

```diff
@@ -171,15 +172,25 @@ public Response chat(@Parameter(description = "human language query", required =
         ChatReply reply = new ChatReply(locale, query);
 
         try {
-            // Check if it's the special OpenNLP implementation with enhanced features
-            if (hli instanceof OpenNLPInterpreter opennlp) {
-                logger.debug("Using OpenNLP interpreter with enhanced features");
-                reply = opennlp.reply(locale, query);
+            Method replyMethod;
+            try {
+                // Check if a public method reply(Locale, String) exists
+                replyMethod = hli.getClass().getMethod("reply", Locale.class, String.class);
```

Comment:
See my comment https://github.com/openhab/openhab-webui/issues/2995#issuecomment-3246912255, this shouldn't be done that way.